### PR TITLE
Add breadcrumbs to help user navigate (fixes #83)

### LINKED ISSUE #83: Add breadcrumbs to help users understand their place in the application
After navigating to this page I have no information on where I am on the application.

ping_page

Why not add a breadcrumb to help us navigate through the app in a proper way. 

something like this 

App / Fenix / Pings / Activation.

### PR DESCRIPTION
**Pull request by DawoodShahat**
This PR solves [https://github.com/mozilla/glean-dictionary/issues/83](https://github.com/mozilla/glean-dictionary/issues/83)

- Breadcrumb added
- Story for Breadcrumb component added

Here is what it looks like.

Screenshot 2020-10-14 at 8 02 23 AM

### CODE REVIEW

**Comment by DawoodShahat** [comment]
@wlach One question, I wasn't able to grab table param when navigated to BigQuery table for a particular ping. The table param isn't added to the params object that is passed as a prop to every component. Take a look when possible.

**Review by wlach — CHANGES_REQUESTED** [review]
This is the right general idea, but I'd like to see a slightly different approach. See comment.

**Comment by DawoodShahat** [comment]
Thanks for the insight. I will add the suggestions.

File: src/App.svelte [hunk 1]
```diff
@@ -11,6 +13,29 @@
 
   let component;
   let params;
+  let links = [];
+
+  afterUpdate(() => {
+    const { app, ping, metric } = params;
+
+    links = [{ url: "/", name: "apps" }];
```
Comment (wlach) [inline src/App.svelte:21]: I'd bet you can simplify this function a bit (i.e. removing a bunch of the if...else) by using the ternary operator (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) and array destructuring (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).

File: stories/breadcrumb.stories.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import Breadcrumb from "../src/components/Breadcrumb.svelte";
+
+const AppsLinks = [{ url: "/", name: "apps" }];
+
+const PingLinks = [
```
Comment (wlach) [inline stories/breadcrumb.stories.js:5]: Again, you could use array destructuring here.

File: src/App.svelte [hunk 3]
```diff
@@ -11,6 +13,29 @@
...
+
+    if (app) {
+      links.push({ url: `/apps/${app}/`, name: app });
+    }
+
+    if (ping) {
+      links.push({
+        url: `/apps/${app}/pings/${ping}`,
+        name: `pings / ${ping}`,
```
Comment (wlach) [inline src/App.svelte:30]: I think we can probably skip the `pings /` part of this.

**Review by wlach — CHANGES_REQUESTED** [review]
This is much closer to what I was expecting, thanks! See comments inline. I think you may have forgotten to add breadcrumbs for the bigquery table pages?

**Comment by DawoodShahat** [comment]
Thanks, @wlach I will work on the suggestions. as for the big query page,  I wasn't able to grab table param when navigated to BigQuery table for a particular ping. The table param isn't added to the params object that is passed as a prop to every component. But I will try to come up with the solution.

**Comment by DawoodShahat** [comment]
Changes made based on @wlach review:

- Updated ```afterUpdate()``` function to avoid unnecessary conditionals
- Updated route for Big Query table, so that its param is exposed through ```params.bigquery``` 
- Added story for ```TableDetail.svelte``` page

File: src/App.svelte (hunk 1 above)
Comment (wlach) [reply src/App.svelte:21]: To be a little more concrete about it, here's what I'd do here:

```js
const links = [
  ...[{ url: "/", name: "apps" }], 
  ...ping ? [{ url: `/apps/${app}/pings/${ping}/`, name: ping }] : [],
  ...metric ? [{ url: `/apps/${app}/metrics/${metric}/`, name: metric }] : [],
];
```

(this example doesn't include BigQuery, leaving that as an exercise for you!)

File: src/App.svelte [hunk 4]
```diff
@@ -20,10 +47,11 @@
   }
 
   page("/", setComponent(AppList));
-  page("/apps/:app/tables/:ping", setComponent(TableDetail));
+  page("/apps/:app/tables/:ping/:bigquery", setComponent(TableDetail));
```
Comment (wlach) [inline src/App.svelte:50]: This should actually be `:table` (using `:ping` was actually a mistake)

```suggestion
  page("/apps/:app/tables/:table", setComponent(TableDetail));
```

File: src/App.svelte [hunk 5]
```diff
@@ -20,10 +47,11 @@
...
   page("/apps/:app/pings/:ping", setComponent(PingDetail));
   page("/apps/:app/metrics/:metric", setComponent(MetricDetail));
   page("/apps/:app", setComponent(AppDetail));
+  page("*", setComponent(Breadcrumb));
```
Comment (wlach) [inline src/App.svelte:54]: Did you mean to add this? It seems like an error and might conflict with some other work that's happening.

File: src/components/Breadcrumb.svelte [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<script>
+  export let links;
+</script>
+
+<style>
+</style>
+
```
Comment (wlach) [inline src/components/Breadcrumb.svelte:7]: You can just skip this section if you're not going to define anything.

```suggestion
```

**Review by wlach — CHANGES_REQUESTED** [review]
This is looking pretty good, code wise! I think it's better if we put the breadcrumbs in the top-left, I actually didn't see them when I first ran your PR. I kind of like the approach I took with Mission Control, which people have generally responded postively to:

image

https://missioncontrol.telemetry.mozilla.org/#/release/windows/main_crashes

(don't worry about reproducing the styling, just the idea is important at this point)

File: src/App.svelte (hunk 5 above)
Comment (DawoodShahat) [reply src/App.svelte:54]: Oh, my bad. I wanted it to be removed.

File: src/App.svelte (hunk 4 above)
Comment (wlach) [reply src/App.svelte:50]: Note to do this you'll also need to make some (very minor) changes to the table component: https://github.com/mozilla/glean-dictionary/blob/9990cef3e5e04a06aaeb6d69470516325e5e8535/src/pages/TableDetail.svelte#L6

**Comment by DawoodShahat** [comment]
Here is a preview

Screenshot 2020-10-21 at 1 46 40 AM

File: src/App.svelte [hunk 7]
```diff
@@ -11,6 +13,36 @@
...
+              url: `/apps/${app}/pings/${table}/`,
+              name: table,
+            },
+            {
+              url: `/apps/${app}/tables/${table}/`,
+              name: `${table} table`,
+            },
+          ]
+        : []),
+    ].filter(Boolean);
```
Comment (wlach) [inline src/App.svelte:44]: I am pretty sure this filter should no longer be necessary?

```suggestion
    ]
```

**Review by wlach — CHANGES_REQUESTED** [review]
This looks great! Just one nit

**Review by wlach — APPROVED** [review]

**Comment by wlach** [comment]
Thank you!

**Pull request merged**